frontend/workflows-lib/lib/components/workflow/TasksFlowUtils.ts

```diff
@@ -61,27 +61,33 @@ export function generateNodesAndEdges(taskNodes: TaskNode[]): {
 } {
   const nodes: Node[] = [];
   const edges: Edge[] = [];
+  const edgeSet = new Set<string>();
 
   const traverse = (tasks: TaskNode[], parents: string[] = []) => {
     tasks.forEach((task) => {
+      const uniqueTaskId = `${task.id}-${task.name}`;
       nodes.push({
-        id: task.id,
+        id: uniqueTaskId,
         type: "custom",
         data: { label: task.name, status: task.status },
         position: { x: 0, y: 0 },
       });
 
       parents.forEach((parent) => {
-        edges.push({
-          id: `e${parent}-${task.id}`,
-          source: parent,
-          target: task.id,
-          animated: true,
-        });
+        const edgeId = `e${parent}-${uniqueTaskId}`;
+        if (!edgeSet.has(edgeId)) {
+          edges.push({
+            id: edgeId,
```

Comment:
Does this need to be a `string` or can it be a tuple?

Reply:
Needs to be string